pkg/cmd/completion/bash_completion.go

```diff
@@ -0,0 +1,188 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package completion
+
+const (
+	// BashCompletionFunc the custom bash completion function to complete object.
+	// The bash completion mechanism will launch the __custom_func func if it cannot
+	// find any completion and pass the object in the ${last_command}
+	// variable. Which then are used to get the pipelineruns/taskruns etc..
+	BashCompletionFunc = `
+__kubectl_get_object()
```

Comment:
Shouldn't we use `tkn` only here ? :angel: (I'm guessing this is inspired from `kubectl` that's why we have the `kubectl` call)

Reply:
it use kubectl to get the object not tkn 😃

Reply:
yes, that's what I think too :angel: we'll need an issue to track that :+1:

Reply:
yeah, no issues with `kubectl`. However `tkn` does support the template output.

Reply:
wdym @hrishin where does it support the template output ?

@vdemeester fyi https://github.com/tektoncd/cli/issues/103

Reply:
@chmouel what do you think of handing over the completion back to `tkn` so that we don't have to rely on `kubectl` or `oc ` to be present?

Reply:
@chmouel try: `tkn t ls -o jsonpath='{.items[].metadata.name}'`

Reply:
ah thanks i didn't know we have this,

Reply:
althoouuughhhhh, we need kubectl still for namespace completion (PR to come)?

Reply:
I have changed it to use tkn instead of kubectl, btu for `-n` `--namespace` i will have to change it :(